html/index.php

```diff
@@ -82,8 +93,16 @@ function newReplyTweet($tweet_textarea, $reply_id) {
           <!--返信課題はここから修正しましょう。-->
           <p><a href = "/index.php?reply=<?= "{$t['id']}" ?>">[返信する]</a>
           <?php if (isset($t['reply_id'])) { ?>
-           <a href="/view.php?id=<?= "{$t['reply_id']}" ?>">[返信元のメッセージ]</a></p>
+            <a href="/view.php?id=<?= "{$t['reply_id']}" ?>">[返信元のメッセージ]</a></p>
+          <?php } ?>
+          <?php if (isMyFavorite($t['id'], $_SESSION['user_id'])) { ?>
+            <a href="/index.php?favorite=<?= "{$t['id']}" ?>"><img class="favorite-image" src='/images/heart-solid-red.svg'></a>
+          <?php } else { ?>
+            <a href="/index.php?favorite=<?= "{$t['id']}" ?>"><img class="favorite-image" src='/images/heart-solid-gray.svg'></a>
           <?php } ?>
+          <?php if (countFavorite($t['id']) > 0) {
+                    echo countFavorite($t['id']);
+                } ?>
```

Comment:
関数countFavoriteはDBを読み込んでいますね。
DBを読み込む処理は不可が大きいので、下記の様に計算結果を変数に入れて省エネにしましょう。

```suggestion
          <?php
          $cnt = countFavorite($t['id']);
          if ($cnt) {
            echo $cnt;
          }
          ?>
```